app/models/role.rb

```diff
@@ -0,0 +1,12 @@
+class Role < ApplicationRecord
+  enum name: {
+    super_admin: "super_admin",
+    admin: "admin",
+    analyst: "analyst"
+  }
+  belongs_to :user
+  belongs_to :resource, polymorphic: true
+  validates :name, presence: true
+
+  scope :super_admin_or_admin, -> { where(name: [:super_admin, :admin]) }
```

Comment:
Because "write" access is essentially the same for an admin and a super_admin, this scope could come in handy.